### PR TITLE
Support custom setting swoole event handle

### PR DESCRIPTION
**Pull request by peibinzhu**
## Update content

Support custom setting swoole event handle, add "MainServerStarting" event.

## Reason

Add "MainServerStarting" event.

Some things need to be done before the swoole server starts. For example, adding a user-defined worker process for monitoring, reporting or other special tasks.

Swoole event callbacks can be defined in the configuration.

Extend swoole event to meet the event callback needs of different scenarios. For example add "onPipeMessage" event for `unixsocket` communication.

## Example

Create a process and send messages to the process periodically.

**config/octane.php**

```php
<?php

use Laravel\Octane\Events\MainServerStarting;
use Laravel\Octane\Swoole\Event;

return [
    //...
    //Ignore other configurations.
    
    /*
    |--------------------------------------------------------------------------
    | Octane Listeners
    |--------------------------------------------------------------------------
    |
    | All of the event listeners for Octane's events are defined below. These
    | listeners are responsible for resetting your application's state for
    | the next request. You may even add your own listeners to the list.
    |
    */

    'listeners' => [
        MainServerStarting::class => [
            App\Listeners\BootProcessListener::class,
        ],
    ],

    /*
    |--------------------------------------------------------------------------
    | Swoole Event Callbacks.
    |--------------------------------------------------------------------------
    |
    | The following settings are custom swoole event callback processing.
    | Note that this will override registered event callback.
    |
    */

    'callbacks' => [
        Event::ON_PIPE_MESSAGE => [App\Bootstrap\PipeMessageCallback::class, 'onPipeMessage'],
    ],
];
```

**app/Listeners/BootProcessListener.php**

```php
<?php

namespace App\Listeners;

use Laravel\Octane\Events\MainServerStarting;
use Swoole\Process as SwooleProcess;

class BootProcessListener
{
    public function handle(object $event): void
    {
        if ($event instanceof MainServerStarting) {
            $server = $event->server;
            $workerCount = $server->setting['worker_num'] + ($server->setting['task_worker_num'] ?? 0) - 1;
            $process = new SwooleProcess(function () use ($server, $workerCount) {
                // Simple example, send messages to the process regularly.
                while (true) {
                    for ($workerId = 0; $workerId <= $workerCount; ++$workerId) {
                        $server->sendMessage('Hello world.', $workerId);
                    }

                    sleep(1);
                }
            }, false, SOCK_DGRAM, true);
            $server->addProcess($process);
        }
    }
}

```

**app/Bootstrap/PipeMessageCallback.php**

```php
<?php

declare(strict_types=1);

namespace App\Bootstrap;

use Swoole\Server;

class PipeMessageCallback
{
    public function onPipeMessage(Server $server, int $fromWorkerId, $data): void
    {
        echo $data . PHP_EOL;
    }
}
```

Run command.

```bash
php artisan octane:start
```

Got the answer:

> Regularly output "Hello world." on the command line.

### CODE REVIEW

**Comment by peibinzhu** [comment]
The above example is just a simple one, it can actually do many things, such as creating user processes, asynchronously collecting request logs or monitoring, etc.

**Comment by taylorotwell** [comment]
A little bit hard to follow. Going to table this for now.
